src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/RazorLanguageServerClient.cs

```diff
@@ -145,11 +162,47 @@ internal class RazorLanguageServerClient : ILanguageClient, ILanguageClientCusto
         var logHubLogger = _loggerProvider.CreateLogger("Razor");
         var razorLogger = new LoggerAdapter(logHubLogger);
         _server = RazorLanguageServerWrapper.Create(serverStream, serverStream, razorLogger, _projectSnapshotManagerDispatcher, ConfigureLanguageServer, _languageServerFeatureOptions);
-
+        await EnsureContainedLanguageServersInitializedAsync();
         var connection = new Connection(clientStream, clientStream);
         return connection;
     }
 
+    private async Task EnsureContainedLanguageServersInitializedAsync()
```

Comment:
Can we call this from the handler, rather than duplicating the code? Or make the logic into a static method or  something at least?

Reply:
The HTMLCSharp server is not allowed to call `_languageClientBroker.LoadAsync` (UI thread contention) but DOES need the list of relevent clients. Here we need to call `LoadAsync` but don't need the clients themselves.

You're right though that we're duplicating. How do you feel about a `GetReleventContainedLanguageServersAndMetadata` and then a function each that actually uses that info in the two locations.

Reply:
Can the client just store the list of relevant clients, as part of initialization, and the handler simply use it directly?

but otherwise, i'm find with your suggestion too. Its the applicability stuff that should be ideally shared